processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -363,34 +386,47 @@ static double estimateSelectivityOfBitmapTree(
    * <p>
    * The non-overlap ratio can be computed like below.
    * <p>
-   * nonOverlapRatio(b1, b2) = size(union(b1, b2)) / (size(b1) + size(b2))
+   * overlapSize = size(b1) + size(b2) - size(union(b1, b2))
+   * nonOverlapRatio(b2) = (size(b2) - overlapSize) / size(b2)
+   * <p>
+   * The approximate unioned size is
+   * <p>
+   * unionedSize = size(b1) + size(b2) * nonOverlapRatio(b2)
    * <p>
    * Given bitmaps, this method calculates the non-overlap ratios of N bitmap samples,
    * and then returns the average of them.
    *
    * @param bitmapIndex   bitmap index to retrieve bitmaps
    * @param bitmapIndexes a list of indexes of bitmaps
    *
-   * @return approximated average non-overlap ratio of bitmaps
+   * @return approximated average non-overlap ratio of bitmaps.
    *
    * @see #estimatePredicateSelectivity(ColumnSelector, String, BitmapIndexSelector, Predicate)
    */
-  private static double computeNonOverlapRatioFromRandomBitmapSamples(
+  static double computeNonOverlapRatioFromRandomBitmapSamples(
```

Comment:
First in your example, non-overlap ratios will be 0.0 and 0.9. They cannot be negatives because ```bSize - overlapSize = unionedSize - preUnionedSize >= 0``` always holds which means ```bSize >= overlapSize```.

I agree on that there will be more efficient and good models for this. There have been a lot of studies on this issue, and we need to investigate them and choose one. However, I'm not sure about completely disabling this feature for multi-value dimensions. Users can just use other strategies until we improve this sufficiently. Isn't it?